pkg/core/transaction/invocation.go

```diff
@@ -35,14 +36,28 @@ func NewInvocationTX(script []byte) *Transaction {
 func (tx *InvocationTX) DecodeBinary(r io.Reader) error {
 	br := util.BinReader{R: r}
 	tx.Script = br.ReadBytes()
-	br.ReadLE(&tx.Gas)
+	if (tx.Version >= 1) {
+		br.ReadLE(&tx.Gas)
+	} else {
+		tx.Gas = util.Fixed8FromInt64(0)
+	}
 	return br.Err
 }
 
 // EncodeBinary implements the Payload interface.
 func (tx *InvocationTX) EncodeBinary(w io.Writer) error {
 	bw := util.BinWriter{W: w}
 	bw.WriteBytes(tx.Script)
-	bw.WriteLE(tx.Gas)
+	if (tx.Version >= 1) {
+		bw.WriteLE(tx.Gas)
+	}
 	return bw.Err
 }
+
+func (tx *InvocationTX) Size() int {
+	sz := util.GetVarSize(tx.Script)
+	if (tx.Version >= 1) {
```

Comment:
Just a thought:
Looks like Version 1+ introduced at some point some incompatible changes to the structures of transactions.
I believe that is could be much better to separate implementations in different packages with have v0 and v1 and having check for version somewhere in one place. Yes in that case we will have dupe code but it will make live easier and cone cleaner. WDYT?
It's just and idea. It we would agree on that -> will create an issue and some day we can do it.

Reply:
That depends on level of similarities (or differences) between versions. Here we have 99% similar code for different versions, so adding new packages will just create a lot of duplication and boilerplate with no real gains, IMO. NEO3 is going to be different, but then I'm not sure if there is a need to keep compatibility with current version when we are to move to NEO3.